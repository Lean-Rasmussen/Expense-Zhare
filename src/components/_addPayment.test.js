//module import
import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
//import components
import AddPaymnet from './addPayment';


const addPayment = shallow(<AddPaymnet />);

it("addPayment Renders", ()=> {
  expect(addPayment).toMatchSnapshot();
})

it("sets the input to state for description",()=>{
  const testText = "enterence to the zoo";
  addPayment.find(".description-input").simulate("change", {target:{value:testText}});
  //expect(addPayment.state().description).toEqual(testText)
})

  it('set the amount input to state iff its a number',()=>{
    const textFailer ="abad";
    const textSuccess = "1513";
    const amountInput = addPayment.find(".amount-input");
    console.log(amountInput)

    it("it fails to set a sting with letters to state",()=>{
      amountInput.simulate("change",{target:{value:textFailer}})
      expect(addPayment.state().expenseAmount).toEqual("john");
    })
    it("lets a numner in stringform to state",()=>{
      amountInput.simulate("change",{target:{value:textSuccess}})
      expect(addPayment.state().expenseAmount).toEqual(textSuccess);
    })
  })
 
