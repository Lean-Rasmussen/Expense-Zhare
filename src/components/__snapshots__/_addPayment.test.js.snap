// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addPayment Renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <addPaymnet />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          Add payment
</h3>,
        <div
          className="addPayment__name-input"
>
          <p>
                    Who paied:
          </p>
          <input
                    className="input"
                    placeholder="Who spend the money?"
                    required={true}
                    type="text"
          />
          <p>
                    Expense description:
          </p>
          <input
                    className="description-input input"
                    onChange={[Function]}
                    placeholder="What did you buy?"
                    required={true}
                    type="text"
                    value=""
          />
          <p>
                    Expense amount:
          </p>
          <input
                    className="amount-input input"
                    min="0"
                    onChange={[Function]}
                    placeholder="How much did you spend?"
                    required={true}
                    type="text"
                    value=""
          />
          <br />
          <button
                    className="btn"
                    onClick={[Function]}
          >
                    Add payment
          </button>
</div>,
      ],
      "className": "addPayment",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Add payment",
        },
        "ref": null,
        "rendered": "Add payment",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              Who paied:
</p>,
            <input
              className="input"
              placeholder="Who spend the money?"
              required={true}
              type="text"
/>,
            <p>
              Expense description:
</p>,
            <input
              className="description-input input"
              onChange={[Function]}
              placeholder="What did you buy?"
              required={true}
              type="text"
              value=""
/>,
            <p>
              Expense amount:
</p>,
            <input
              className="amount-input input"
              min="0"
              onChange={[Function]}
              placeholder="How much did you spend?"
              required={true}
              type="text"
              value=""
/>,
            <br />,
            <button
              className="btn"
              onClick={[Function]}
>
              Add payment
</button>,
          ],
          "className": "addPayment__name-input",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Who paied:",
            },
            "ref": null,
            "rendered": "Who paied:",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "input",
              "placeholder": "Who spend the money?",
              "required": true,
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Expense description:",
            },
            "ref": null,
            "rendered": "Expense description:",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "description-input input",
              "onChange": [Function],
              "placeholder": "What did you buy?",
              "required": true,
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Expense amount:",
            },
            "ref": null,
            "rendered": "Expense amount:",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "amount-input input",
              "min": "0",
              "onChange": [Function],
              "placeholder": "How much did you spend?",
              "required": true,
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Add payment",
              "className": "btn",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Add payment",
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Add payment
</h3>,
          <div
            className="addPayment__name-input"
>
            <p>
                        Who paied:
            </p>
            <input
                        className="input"
                        placeholder="Who spend the money?"
                        required={true}
                        type="text"
            />
            <p>
                        Expense description:
            </p>
            <input
                        className="description-input input"
                        onChange={[Function]}
                        placeholder="What did you buy?"
                        required={true}
                        type="text"
                        value=""
            />
            <p>
                        Expense amount:
            </p>
            <input
                        className="amount-input input"
                        min="0"
                        onChange={[Function]}
                        placeholder="How much did you spend?"
                        required={true}
                        type="text"
                        value=""
            />
            <br />
            <button
                        className="btn"
                        onClick={[Function]}
            >
                        Add payment
            </button>
</div>,
        ],
        "className": "addPayment",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add payment",
          },
          "ref": null,
          "rendered": "Add payment",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                Who paied:
</p>,
              <input
                className="input"
                placeholder="Who spend the money?"
                required={true}
                type="text"
/>,
              <p>
                Expense description:
</p>,
              <input
                className="description-input input"
                onChange={[Function]}
                placeholder="What did you buy?"
                required={true}
                type="text"
                value=""
/>,
              <p>
                Expense amount:
</p>,
              <input
                className="amount-input input"
                min="0"
                onChange={[Function]}
                placeholder="How much did you spend?"
                required={true}
                type="text"
                value=""
/>,
              <br />,
              <button
                className="btn"
                onClick={[Function]}
>
                Add payment
</button>,
            ],
            "className": "addPayment__name-input",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Who paied:",
              },
              "ref": null,
              "rendered": "Who paied:",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "input",
                "placeholder": "Who spend the money?",
                "required": true,
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Expense description:",
              },
              "ref": null,
              "rendered": "Expense description:",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "description-input input",
                "onChange": [Function],
                "placeholder": "What did you buy?",
                "required": true,
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Expense amount:",
              },
              "ref": null,
              "rendered": "Expense amount:",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "amount-input input",
                "min": "0",
                "onChange": [Function],
                "placeholder": "How much did you spend?",
                "required": true,
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Add payment",
                "className": "btn",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Add payment",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
